### PR TITLE
chore: use eslint rule to avoid relative imports through parent [WEB-1496]

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

Adds an eslint plugin and rule, which on `make fmt` updates our existing relative imports to absolute paths
Example `../Avatar` to `components/Avatar`

## Test Plan

- Codebase passes `make fmt` and `npm run build`
- Satisfactory updates to imports

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by ashtonG** [comment]
we should have this rule from the `eslint-plugin-import` package already in the import/no-relative-parent-imports rule: https://github.com/import-js/eslint-plugin-import/blob/HEAD/docs/rules/no-relative-parent-imports.md

**Comment by johnkim-det** [comment]
we discussed that `no-relative-parent-imports` looks like it's more restrictive than we actually want, which is why `eslint-plugin-no-relative-import-paths` was used.

however, I also came across [this issue comment](https://github.com/import-js/eslint-plugin-import/issues/1392#issuecomment-824222201), which suggests using a `no-restricted-imports` rule instead. Seem like this may be a good solution that doesn't requiring adding a plug-in package.

**Comment by mapmeld** [comment]
> [this issue comment](https://github.com/import-js/eslint-plugin-import/issues/1392#issuecomment-824222201), which suggests using a `no-restricted-imports` rule instead

oh, that's a good config for the rule, I agree it's better to implement this with the built-in if that pattern works

**Comment by mapmeld** [comment]
Applied the new rule, rebased, and confirmed it caught a new parent-relative-import which needed to be reformatted

File: webui/react/src/components/UPlot/UPlotScatter.tsx [hunk 1]
```diff
@@ -4,8 +4,7 @@ import uPlot from 'uplot';
 import UPlotChart, { Options } from 'components/UPlot/UPlotChart';
 import QuadTree, { pointWithin } from 'components/UPlot/UPlotScatter/quadtree';
 import { Range } from 'types';
-
-import { Scale } from '../../types';
+import { Scale } from 'types';
```
Comment (ashtonG) [inline webui/react/src/components/UPlot/UPlotScatter.tsx:7]: can we merge imports like this into a single line?

Comment (mapmeld) [reply webui/react/src/components/UPlot/UPlotScatter.tsx]: found and updated 3 files -- added a ticket to catch other files with a no-duplicate-imports rule

**Pull request merged**